model/CSVDialog.cc

```diff
@@ -138,7 +140,25 @@ std::string CSVDialog::loadWebFile(const std::string& url)
   const std::string tempStr    = temp.string();
           
   std::ofstream outFile(tempStr, std::ofstream::out);  
-  outFile << res.get().body();                                            
+ 
+  // Deal with zipped csv files 
+  std::string targetStr = target.str();
+  std::string fileExt = ".zip";
+  std::size_t found = targetStr.find(fileExt);
+  
+  if (found!=std::string::npos) {
+      vector<unsigned char> zbuf(res.get().body().size());
+      DeflateZStream zs(res.get().body(), zbuf);
+      zs.deflate();  
+      
+      vector<char> cbuf(a85::size_for_a85(zs.total_out,false));
+      a85::to_a85(&zbuf[0],zs.total_out, &cbuf[0], false);
+      // this ensures that the escape sequence ']]>' never appears in the data
+      replace(cbuf.begin(),cbuf.end(),']','~');      
+    
```

Comment:
What is this code doing in CSVDialog? a85 is about ASCII representation of binary data, which is not required for handling zipped CSV files.